pkg/config/load.go

```diff
@@ -0,0 +1,114 @@
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+
+	cioperatorapi "github.com/openshift/ci-operator/pkg/api"
+)
+
+func readCiOperatorConfig(configFilePath string) (*cioperatorapi.ReleaseBuildConfiguration, error) {
+	data, err := ioutil.ReadFile(configFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read ci-operator config (%v)", err)
+	}
+
+	var configSpec *cioperatorapi.ReleaseBuildConfiguration
+	if err := yaml.Unmarshal(data, &configSpec); err != nil {
+		return nil, fmt.Errorf("failed to load ci-operator config (%v)", err)
+	}
+
+	if err := configSpec.Validate(); err != nil {
+		return nil, fmt.Errorf("invalid ci-operator config: %v", err)
+	}
+
+	return configSpec, nil
+}
+
+// path to ci-operator configuration file encodes information about tested code
+// .../$ORGANIZATION/$REPOSITORY/$BRANCH.$EXT
```

Comment:
Total-nit: The comment of course rot. Let's update (it's now something like `...$ORG/$REPO/$ORG-$REPO-$BRANCH{,__$VARIANT}.yaml`) or delete it